app/src/main/java/de/westnordost/streetcomplete/quests/max_height/AddMaxHeightForm.java

```diff
@@ -0,0 +1,196 @@
+	}
+
+	private void addOtherAnswers()
+	{
+
+		if (getCountryInfo().getHeightUnits().size() == 2)
+		{
+			if (isMetric)
+			{
+				addOtherAnswer(R.string.quest_maxheight_answer_imperial_unit, () ->
+				{
+					isMetric = false;
+					setMaxHeightSignLayout(getMaxHeightLayoutResourceId());
+				});
+			}
+			else
+			{
+				addOtherAnswer(R.string.quest_maxheight_answer_metric_unit, () ->
+				{
+					isMetric = true;
+					setMaxHeightSignLayout(getMaxHeightLayoutResourceId());
+				});
+			}
+		}
+
+		addOtherAnswer(R.string.quest_maxheight_answer_noSign, () ->
+		{
+			new AlertDialogBuilder(getActivity())
+				.setTitle(R.string.quest_maxheight_answer_noSign_confirmation_title)
+				.setMessage(R.string.quest_maxheight_answer_noSign_confirmation)
+				.setPositiveButton(R.string.quest_maxheight_answer_noSign_confirmation_positive, (dialog, which) -> {
+					Bundle answer = new Bundle();
+					answer.putBoolean(NO_SIGN, true);
+					applyImmediateAnswer(answer);
+				})
+				.setNegativeButton(R.string.quest_generic_confirmation_no, null)
+				.show();
+		});
+	}
+
+	@Override protected void onClickOk()
+	{
+		if(!hasChanges())
+		{
+			Toast.makeText(getActivity(), R.string.no_changes, Toast.LENGTH_SHORT).show();
+			return;
+		}
+
+		if(userSelectedUnrealisticHeight())
+		{
+			confirmUnusualInput(this::applyMaxHeightFormAnswer);
+		}
+		else
+		{
+			applyMaxHeightFormAnswer();
+		}
+	}
+
+	private boolean userSelectedUnrealisticHeight()
+	{
+		double height = Double.parseDouble(getHeightFromInput());
+		double heightInMeter = isMetric ? height : feetToMeter(height);
+		return heightInMeter > 6;
+	}
+
+	private static double feetToMeter(double feet)
+	{
+		return feet / 3.2808;
+	}
+
+	private void applyMaxHeightFormAnswer()
+	{
+		Bundle answer = new Bundle();
+
+		String height = getFinalMaxHeight(getHeightFromInput());
+
+		// metric is the OSM default, does not need to be mentioned
+		if(!isMetric)
+		{
+			//this adds an apostrophe and a double-quote to be in a format like e.g. 6'7"
+			height = height.replace(".", "'");
+			height += "\"";
+		}
+		answer.putString(MAX_HEIGHT, height);
+
+		applyFormAnswer(answer);
+	}
+
+	private String getFinalMaxHeight(String value)
+	{
+		if (isMetric)
+		{
+			/*TODO: should a trailing zero be added or does it even work without?
+			https://taginfo.openstreetmap.org/keys/maxheight#values contains both...*/
+			if (!value.contains("."))
+			{
+				value += ".0";
+			}
+			return value;
+		}
+		else
+		{
+			return value.replace(".", "'") + "\"";
+		}
+	}
+
+	private String getHeightFromInput()
+	{
+		if (isMetric)
+		{
+			if (!heightInput.getText().toString().isEmpty())
+			{
+
+				return heightInput.getText().toString().replace(",", ".");
+			}
+			else
+			{
+				return null;
+			}
+		}
+		else
+		{
+			if (!feetInput.getText().toString().isEmpty() && !inchInput.getText().toString().isEmpty())
+			{
+				return feetInput.getText().toString() + "." + inchInput.getText().toString();
```

Comment:
Although simplifying inches to a range from 12 to 10 doesn't seem so bad, an 11 inch measurement will give an odd result. (I admittedly didn't check against boundary condition.)